wgpu-hal/src/gles/queue.rs

```diff
@@ -0,0 +1,534 @@
+                        start_vertex as i32,
+                        vertex_count as i32,
+                        instance_count as i32,
+                    );
+                }
+            }
+            C::DrawIndexed {
+                topology,
+                index_type,
+                index_count,
+                index_offset,
+                base_vertex,
+                instance_count,
+            } => match (base_vertex, instance_count) {
+                (0, 1) => gl.draw_elements(
+                    topology,
+                    index_count as i32,
+                    index_type,
+                    index_offset as i32,
+                ),
+                (0, _) => gl.draw_elements_instanced(
+                    topology,
+                    index_count as i32,
+                    index_type,
+                    index_offset as i32,
+                    instance_count as i32,
+                ),
+                (_, 1) => gl.draw_elements_base_vertex(
+                    topology,
+                    index_count as i32,
+                    index_type,
+                    index_offset as i32,
+                    base_vertex,
+                ),
+                (_, _) => gl.draw_elements_instanced_base_vertex(
+                    topology,
+                    index_count as _,
+                    index_type,
+                    index_offset as i32,
+                    instance_count as i32,
+                    base_vertex,
+                ),
+            },
+            C::DrawIndirect {
+                topology: _,
+                indirect_buf,
+                indirect_offset: _,
+            } => {
+                gl.bind_buffer(glow::DRAW_INDIRECT_BUFFER, Some(indirect_buf));
+                //TODO: https://github.com/grovesNL/glow/issues/172
+                //gl.draw_arrays_indirect(topology, indirect_offset);
+            }
+            C::DrawIndexedIndirect {
+                topology: _,
+                index_type: _,
+                indirect_buf,
+                indirect_offset: _,
+            } => {
+                gl.bind_buffer(glow::DRAW_INDIRECT_BUFFER, Some(indirect_buf));
+                //TODO: https://github.com/grovesNL/glow/issues/172
+            }
+            C::Dispatch(group_counts) => {
+                gl.dispatch_compute(group_counts[0], group_counts[1], group_counts[2]);
+            }
+            C::DispatchIndirect {
+                indirect_buf,
+                indirect_offset,
+            } => {
+                gl.bind_buffer(glow::DRAW_INDIRECT_BUFFER, Some(indirect_buf));
+                gl.dispatch_compute_indirect(indirect_offset as i32);
+            }
+            C::FillBuffer { .. } => unimplemented!(),
+            C::CopyBufferToBuffer {
+                src,
+                src_target,
+                dst,
+                dst_target,
+                copy,
+            } => {
+                gl.bind_buffer(src_target, Some(src));
```

Comment:
`WebGL` related fix no longer required ?

https://github.com/gfx-rs/gfx/blob/d51857608dff4359e80a61057bca121f476e482e/src/backend/gl/src/queue.rs#L621-L665

Reply:
We'll probably need it, still. But at this point I'm basically re-implementing the backend, and I'm trying to focus on GLES3 first. After it lands, we'll follow up with fixes needed for WebGL2 specifically.